### PR TITLE
[3879] refactor SyncManager & EventHandlers

### PR DESCRIPTION
**Pull request by kanat**
### 🎯 Goal

- Make `SyncManager` & `EventHandler` independent. 
- Remove `EventHandlerProvider` and let Plugin resolve dependencies
- Fix issues in `SyncManager`:
   - start syncing as soon as possible 
   - use `cids` from `SyncState` to restore
   - do not call `updateActiveChannels` if `updateActiveQueryChannels` fails

### 🧪 Testing

- Run any sample app
- Check the following scenarios:
   - resending the failed messages
   - backgrounding / foregrounding
   - enabling / disabling airplane mode

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

**Comment by leandroBorgesFerreira** [comment]
I did not understand the difference between:
```
 val childResolver: DependencyResolver
```
and:
```
val provideDependency: (KClass<*>) -> Any?
```
It would be nice to cover this in the documentation of the PR or in Kdocs

**Review by adasiewiczr — COMMENTED** [review]
Great job! Left some comments (mostly KDocs related)

File: stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/event/handler/internal/EventHandlerImpl.kt [hunk 1]
```diff
@@ -679,23 +629,22 @@ internal class EventHandlerImpl(
     }
 
     private fun Message.enrichWithOwnReactions(batch: EventBatchUpdate, user: User?) {
-        ownReactions = if (user != null && clientMutableState.user.value?.id != user.id) {
+        ownReactions = if (user != null && currentUserId != user.id) {
             batch.getCurrentMessage(id)?.ownReactions ?: mutableListOf()
         } else {
             mergeReactions(
-                latestReactions.filter { it.userId == (clientMutableState.user.value?.id ?: "") },
+                latestReactions.filter { it.userId == (currentUserId ?: "") },
```
[the remark this answers is not in the document]
Comment (kanat) [reply stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/event/handler/internal/EventHandlerImpl.kt:636]: Good catch, thanks

**Review by leandroBorgesFerreira — APPROVED** [review]
Looks good to me 🚢

**Pull request merged**